### DISCUSSION TITLE
retrieve field and add to state

### DISCUSSION THREAD

**Discussion by Paul75** [k0shk0sh/ComposeEasyForms#13] [Q&A] [answered]
Helllo,

I want to retrieve field and put to state. I do like : 

```
BuildAddressForm(navController, product) { it ->
        // errorsState = it.formData().toString()
        val ValuesForm: List<EasyFormsResult> = it.formData()

        // mise en state
        val addressToSave = Address("", "", "", "", postalCode = "", city = "") // error

        ValuesForm.forEach { it2: EasyFormsResult ->
            // addressToSave[it2.key] = it2.value
// not works
        }

        viewModel.buy_updateAddress(addressState = addressToSave)

        navController.navigate(BottomNavigationScreens.ProductBuy.route + "/" + product.id)
    }
```

```
data class Address(
    val name: String,
    val country: String,
    val address: String,
    val address2: String? = null,
    val postalCode: String,
    val city: String
)
```

But errors in syntaxe ....
Thanks for help ...

**Comment by k0shk0sh** [comment]
You should call `it.formData()` only on button click. Please refer to example app for your reference.

**Reply by Paul75** [reply]
Hello,

I see exemple not understand when I get Eelement into resultForm ?

In can forEach into List<EasyFormsResult> but after only have it.key and not it.value ?

Thanks

**Comment by Paul75** [comment]
Hello,

I not recupere the value it say .value not exist ???

I do a function like : 

```
private fun searchInFormToAdress(easyformsAddressDatas: List<EasyFormsResult>, item: String): String {
        easyformsAddressDatas.forEach {
            Log.d("searchInFormToAdress", it.toString())
            Log.d("searchInFormToAdress", it.key.toString())
            Log.d("searchInFormToAdress", it.value) <---- it.value not found

        }
        return "ZZZ"
    }
```

Thanks for help

**Comment by Paul75** [comment]
in the doc it see the value parameter but in the foreach only key and easyFormsErrorState ...

https://k0shk0sh.github.io/ComposeEasyForms/easyforms/com.github.k0shk0sh.compose.easyforms/-easy-forms-result/-string-result/index.html

**Reply by k0shk0sh** [reply]
Please follow this https://github.com/k0shk0sh/ComposeEasyForms/blob/main/app/src/main/java/com/github/k0shk0sh/compose/easyforms/example/MainViewModel.kt#L12

**Reply by Paul75** [reply]
Hello,

Yes works I found yersterday

Sorry and Thanks for your works

**Comment by k0shk0sh** [comment, marked as answer]
Please follow this https://github.com/k0shk0sh/ComposeEasyForms/blob/main/app/src/main/java/com/github/k0shk0sh/compose/easyforms/example/MainViewModel.kt#L12
